### ISSUE TITLE
1

### ISSUE THREAD

**Issue by YeRen123455** [YeRen123455/Infrared-Small-Target-Detection#11]

**Comment by YeRen123455** [comment]
Hi @jbeomlee93
Thanks for your reply. I have followed your suggestion in #7 (comment), #8 (comment) and retrained the "SEAM+Adv_cam". But the performance is even worse. I have changed 4 places of your "obtain_CAM_masking.py" to reproduce the "SEAM+Adv_cam". The changes are as follows:

[ 1] I followed your reply in combine with seam Can you provide your trained weights for MDvsFA-cGAN? #7 (comment) and set the masking threshold $T$, regularization item $\lambda$ as 8 and 2, respectively. That is:
       parser.add_argument("--AD_coeff",    default=2, type=int)
       parser.add_argument("--score_th",    default=8, type=float)     in  "obtain_CAM_masking.py"
[ 2] I tried to obey your suggestion in Question about Table 4. 文章中的错误 #8 (comment), but if the adversarial climbing was done on logit, GAP(cam), before up-sampled and before PCM module. It means I should output the cam value in forward function defined in "resnet38_SEAM.py". That is：
      def forward(self, x, separate=False):
      N, C, H, W = x.size()
      d = super().forward_as_dict(x)
      cam = self.fc8(self.dropout7(d['conv6']))
      if separate:
      return cam in "resnet38_SEAM.py"
[ 3] At the same time, I should also accumulate our localization maps (in Equation 4) using CAM after PCM module. It means I should change the code of "resnet38_SEAM.py" as:
  def forward(self, x, separate=False):
  N, C, H, W = x.size()
  d = super().forward_as_dict(x)
  cam = self.fc8(self.dropout7(d['conv6']))
  ............
  ............
  cam_rv_1 = self.PCM(cam_d_norm, f)
  if separate:
  return cam_rv_1 in "resnet38_SEAM.py"
However, the step 2 and 3 are contradictory. I can only output cam or cam_rv_1. Otherwise I can not get the value of "regions" in your "obtain_CAM_masking.py" by grad-cam (that is because I can not get the model's gradient if I output both cam and cam_rv_1).

[ 4] Since the CAM output of SEAM has 21 classes (20 classes + background), the default class label for your adv_cam is 20. I change the "gradCAM.py" as follows:
  self.logits = self. model(image, separate= True)[ :, 1:, :, :] in "gradCAM.py"
The above changes can not make me to successfully reproduce SA. Could you please help me to check these changes. I really want to follow your work! Thanks a lot!

**Comment by YeRen123455** [comment]
Hi @jbeomlee93
Thanks for your reply. I have followed your suggestion in #7 (comment), #8 (comment) and retrained the "SEAM+Adv_cam". But the performance is even worse. I have changed 4 places of your "obtain_CAM_masking.py" to reproduce the "SEAM+Adv_cam". The changes are as follows:

[ 1] I followed your reply in combine with seam Can you provide your trained weights for MDvsFA-cGAN? #7 (comment) and set the masking threshold $T$, regularization item $\lambda$ as 8 and 2, respectively. That is:
```
       parser.add_argument("--AD_coeff",    default=2, type=int)
       parser.add_argument("--score_th",    default=8, type=float)     in  "obtain_CAM_masking.py"
```
[ 2] I tried to obey your suggestion in Question about Table 4. 文章中的错误 #8 (comment), but if the adversarial climbing was done on logit, GAP(cam), before up-sampled and before PCM module. It means I should output the cam value in forward function defined in "resnet38_SEAM.py". That is：
      def forward(self, x, separate=False):
      N, C, H, W = x.size()
      d = super().forward_as_dict(x)
      cam = self.fc8(self.dropout7(d['conv6']))
      if separate:
      return cam in "resnet38_SEAM.py"
[ 3] At the same time, I should also accumulate our localization maps (in Equation 4) using CAM after PCM module. It means I should change the code of "resnet38_SEAM.py" as:
  def forward(self, x, separate=False):
  N, C, H, W = x.size()
  d = super().forward_as_dict(x)
  cam = self.fc8(self.dropout7(d['conv6']))
  ............
  ............
  cam_rv_1 = self.PCM(cam_d_norm, f)
  if separate:
  return cam_rv_1 in "resnet38_SEAM.py"
However, the step 2 and 3 are contradictory. I can only output cam or cam_rv_1. Otherwise I can not get the value of "regions" in your "obtain_CAM_masking.py" by grad-cam (that is because I can not get the model's gradient if I output both cam and cam_rv_1).

[ 4] Since the CAM output of SEAM has 21 classes (20 classes + background), the default class label for your adv_cam is 20. I change the "gradCAM.py" as follows:
  self.logits = self. model(image, separate= True)[ :, 1:, :, :] in "gradCAM.py"
The above changes can not make me to successfully reproduce SA. Could you please help me to check these changes. I really want to follow your work! Thanks a lot!
